### ISSUE TITLE
base_score - allow to specify as in xgboost

### ISSUE THREAD

**Issue by gugatr0n1c** [Microsoft/LightGBM#78]
Hi,

in xgboost there is setting for base_score.
It represents 0th value of prediction, before first tree is built.

In LightGMB it seems to me this value is hardcoded to 0.

But for regression task this is not good choice (even for classification is better 0.5 as in xgboost).
Suppose that your mean of target is something big and you need learning_rate small, then thousands of trees are just catching this mean and are actually not very useful.

1] It would be great to have choice to specify this value.

2] default value should be "intelligent" as mean(learning_target) - this bring out of box smaller need of number of trees, so even better speed. From my experience without any influence on accuracy.
- but for some task median is better, that is way I propose 1] and 2]

thx for consideration

**Comment by wxchan** [comment]
It actually can be solved with initial score now. Use a file with same score * shape of data. It will be even easier when you are using python/R package in the future.

**Comment by gugatr0n1c** [comment]
great! my mistake not reading docs carefully, thank you
